tekton/resources/prow-config-cd.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: trigger-to-deploy-prow-config
+spec:
+  params:
+  - name: gitRepository
+    value: $(body.gitRepository)
+  - name: gitRevision
+    value: $(body.gitRevision)
+  - name: configPath
+    value: $(body.configPath)
+  - name: namespace
+    value: $(body.namespace)
+  - name: clusterResource
+    value: $(body.clusterResource)
+---
+apiVersion: tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: prow-config-deployer
+spec:
+  triggers:
+    - name: trigger
+      binding:
+        name: trigger-to-deploy-prow-config
+      template:
+        name: deploy-prow-config
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: deploy-prow-config
+spec:
+  params:
+  - name: gitRepository
+    description: URL of the repository that holds the prow configuration
+  - name: gitRevision
+    description: Git revision
+  - name: namespace
+    description: Namespace to deploy to in the target cluster
+  - name: configPath
+    description: Path in the git repo that holds prow configs
+  - name: clusterResource
+    description: Name of the cluster resource that points to the target cluster
+  resourcetemplates:
+  - apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: git-source-$(uid)
+    spec:
+      type: git
+      params:
+      - name: revision
+        value: $(params.gitRevision)
+      - name: url
+        value: https://$(params.gitRepository)
+  - apiVersion: tekton.dev/v1alpha1
+    kind: TaskRun
+    metadata:
+      generateName: deploy-prow-config-$(uid)-
+    spec:
+      taskSpec:
+        inputs:
+          params:
+          - name: configPath
+            description: Path in the git repo that holds prow configs
+          - name: namespace
+            description: Namespace to deploy to in the target cluster
+          resources:
+          - name: source
+            type: git
+          - name: prow-cluster
+            type: cluster
+        steps:
+        - name: deploy
+          image: [image with kubectl]
+          command:
+          - /bin/sh
+          args:
+          - -ce
+          - |
+            kubectl create configmap config \
+              --from-file=$(inputs.resources.source.path)/$(inputs.params.configPath)=prow/config.yaml \
+              --dry-run -o yaml | \
+              kubectl replace configmap config -n $(inputs.params.namespace) -f -
+          env:
+            name: KUBECONFIG
+            value: /workspace/$(inputs.resources.prow-cluster.name)/kubeconfig
+        - name: verify
+          image: [image with kubectl]
+          command:
+          - /bin/sh
+          args:
+          - -ce
+          - |
+            # Check that prow is up and running
```

Comment:
Should be "check that prow picks up the new configuration" :sweat_smile: 
Except parsing the tide/hook logs I have no idea how to validate this… :sweat:

Reply:
Yeah, not sure how to do that... it really depends on how prow would react to bad configs